### PR TITLE
IO of OpenMesh files - use `double`s instead of `float`s

### PR DESCRIPTION
**Pull request by janetournois**
## Summary of Changes

The default OpenMesh point type is made of `float`s. This PR changes the traits we use in CGAL to you `double`s and ensure that reading an off/writing an om/reading om gives the same result back.
This PR completes #8427

## Release Management

* Affected package(s): BGL, Demo
* License and copyright ownership: unchanged

### CODE REVIEW

File: BGL/include/CGAL/boost/graph/IO/OM.h [hunk 1]
```diff
@@ -108,7 +109,7 @@ bool write_OM(std::string fname, const Graph& g, VPM vpm, VFeaturePM vfpm, EFeat
     omesh.status(omv).set_feature(isfeature);
   }
 
-  return OpenMesh::IO::write_mesh(omesh, fname, OpenMesh::IO::Options::Status);
+  return OpenMesh::IO::write_mesh(omesh, fname, OpenMesh::IO::Options::Status, 18);
```
[the remark this answers is not in the document]
Comment (janetournois) [reply BGL/include/CGAL/boost/graph/IO/OM.h:112]: absolutely!
Thanks

File: Lab/demo/Lab/Plugins/IO/OM_io_plugin.cpp [hunk 2]
```diff
@@ -180,12 +180,14 @@ save(QFileInfo fileinfo, QList<CGAL::Three::Scene_item*>& items)
     res = CGAL::IO::write_OM((const char*)fileinfo.filePath().toUtf8()
               , *sm_item->face_graph()
               , CGAL::parameters::vertex_is_constrained_map(selection_item->constrained_vertices_pmap())
-              .edge_is_constrained_map(selection_item->constrained_edges_pmap()));
+              .edge_is_constrained_map(selection_item->constrained_edges_pmap())
+              .stream_precision(18));
```
Comment (sloriot) [inline Lab/demo/Lab/Plugins/IO/OM_io_plugin.cpp:184]:
```suggestion
              .stream_precision(17));
```

File: Lab/demo/Lab/Plugins/IO/OM_io_plugin.cpp [hunk 3]
```diff
@@ -180,12 +180,14 @@ save(QFileInfo fileinfo, QList<CGAL::Three::Scene_item*>& items)
...
   }
   else
   {
     res = CGAL::IO::write_OM((const char*)fileinfo.filePath().toUtf8()
-              , *sm_item->face_graph());
+              , *sm_item->face_graph()
+              , CGAL::parameters::stream_precision(18));
```
Comment (sloriot) [inline Lab/demo/Lab/Plugins/IO/OM_io_plugin.cpp:190]:
```suggestion
              , CGAL::parameters::stream_precision(17));
```

**Comment by sloriot** [comment]
Successfully tested in CGAL-6.1-Ic-22

File: BGL/include/CGAL/boost/graph/IO/OM.h [hunk 4]
```diff
@@ -228,7 +235,9 @@ bool write_OM(const std::string& fname,
                                CGAL::Constant_property_map<edge_descriptor, bool>(false));
   auto vpm = choose_parameter(get_parameter(np, internal_np::vertex_point),
                               get_const_property_map(vertex_point, g));
-  return internal::write_OM(fname, g, vpm, vfpm, efpm);
+  std::streamsize precision = choose_parameter(get_parameter(np, internal_np::stream_precision),
+                                               18);
```
Comment (lrineau) [inline BGL/include/CGAL/boost/graph/IO/OM.h:239]: `18` is too much for a default. The value `std::numeric_limits<double>::max_digits10` is equal to 17.

Besides, the documentation says:
>  **Default**: the precision of the stream `os`
but:
  - there is no stream `os` in the parameter of the function
  - the default for a new `std::ofstream` is 6.

In my opinion, the default should be `17` and documented as such.

Comment (afabri) [reply BGL/include/CGAL/boost/graph/IO/OM.h:239]: Just drop the word `os` and replace 18 with 17  directly in `main` with your super powers.

Comment (lrineau) [reply BGL/include/CGAL/boost/graph/IO/OM.h:239]: Done in https://github.com/CGAL/cgal/commit/6cf3b104038c678a51b36ace59361eb62b2b85c5

**Pull request merged**